arduino-ide-extension/src/node/core-service-impl.ts

```diff
@@ -201,14 +205,48 @@ export class CoreServiceImpl extends CoreClientAware implements CoreService {
     ) => (request: REQ) => ClientReadableStream<RESP>,
     errorCtor: ApplicationError.Constructor<number, CoreError.ErrorLocation[]>,
     task: string
-  ): Promise<void> {
+  ): Promise<ApiUploadResponse> {
+    const uploadResponseFragment: Mutable<Partial<ApiUploadResponse>> = {
+      portBeforeUpload: options.port,
+      portAfterUpload: options.port, // assume no port changes during the upload
+    };
     const coreClient = await this.coreClient;
     const { client, instance } = coreClient;
     const progressHandler = this.createProgressHandler(options);
-    const handler = this.createOnDataHandler(progressHandler);
+    // Track responses for port changes:
+    //  - No port changes are expected when uploading using a programmer.
+    //  - When the `updatedUploadPort` is missing, the port did not change.
```

Comment:
A change was made to https://github.com/arduino/arduino-cli/pull/2253 that will result in the `updated_upload_port` field now always being present in the [cc.arduino.cli.commands.v1.ArduinoCoreService.Upload](https://arduino.github.io/arduino-cli/dev/rpc/commands/#cc.arduino.cli.commands.v1.Port)
response message:

https://github.com/arduino/arduino-cli/pull/2253/commits/5896fba900db441a1a3a2a3af30536c0d8cb3425

So it should now be technically possible to remove from the Arduino IDE codebase all code for handling the absence of that field. I'm not sure whether you'll prefer to still leave the code in as a "belt and suspenders" measure.